tests/test_globalization.py

```diff
@@ -3,91 +3,148 @@
         """
         local_machine_name = socket.gethostname().replace('-','_').replace('.','_')
         try:
             client = create_mssql_cli_client()
-            max_string_length = 50
 
-            # Takes 50 characters at a time from charset until charset becomes empty.
-            # Each time, the characters are used for 'create table' and 'insert into' statement
+            # Each characters in characters_list is a string with max length 50
+            # Each time in the for loop, the string used for 'create table' and 'insert into' statement
             # that are executed by client.execute_query().
             # We validates the query results are the same value we inserted and
             # they are properly unicode encoded.
-            for charset in charset_list:
-                while len(charset) > 0:
-                    test_string_length = min(len(charset), max_string_length)
-                    test_str = charset[0:test_string_length]
-                    charset = charset[test_string_length:]
-                    col1_name = u'col_{0}1'.format(test_str)
-                    col2_name = u'col_{0}2'.format(test_str)
-                    table_name = u'#mssqlcli_{0}_{1}_{2}'.format(local_machine_name, random_str(), test_str)
-                    setup_query = u"CREATE TABLE {0} ({1} nvarchar(MAX), {2} int);"\
-                        u"INSERT INTO {0} VALUES (N'value_{3}1', 1);"\
-                        u"INSERT INTO {0} VALUES (N'value_{3}2', 2);"\
-                        .format(table_name, col1_name, col2_name, test_str)
-                    
-                    for rows, columns, status, statement, is_error in client.execute_query(setup_query):
-                        assert is_error == False
-
-                    select_query = u"SELECT {0}, {1} FROM {2};".format(col1_name, col2_name, table_name)
-                    for rows, columns, status, statement, is_error in client.execute_query(select_query):
-                        assert is_error == False
-                        assert len(rows) == 2
-                        assert decode(rows[0][0]) == u'value_{0}1'.format(test_str)
-                        assert decode(rows[1][0]) == u'value_{0}2'.format(test_str)
+            for characters in characters_list:
+                test_str = characters
+                col1_name = u'col_{0}1'.format(test_str)
+                col2_name = u'col_{0}2'.format(test_str)
+                table_name = u'#mssqlcli_{0}_{1}_{2}'.format(local_machine_name, random_str(), test_str)
+                setup_query = u"CREATE TABLE {0} ({1} nvarchar(MAX), {2} int);"\
+                    u"INSERT INTO {0} VALUES (N'value_{3}1', 1);"\
+                    u"INSERT INTO {0} VALUES (N'value_{3}2', 2);"\
+                    .format(table_name, col1_name, col2_name, test_str)
+                
+                for rows, columns, status, statement, is_error in client.execute_query(setup_query):
```

Comment:
For loop needed?

Reply:
unfortunately yes. client.execute_query(query) is a generator.

Reply:
dang. we need to add execute_non_query and execute_scalar methods some day.

Reply:
yup :)